### ISSUE TITLE
Error 500 on showing note with no comments

### ISSUE THREAD

**Issue by westnordost** [openstreetmap/openstreetmap-website#2146]
### Observed behavior
Displaying a note with no comments yields an error 500.

### How to reproduce
Display this note: https://www.openstreetmap.org/note/940980
the note via the API: https://api.openstreetmap.org/api/0.6/notes/940980

### Expected behavior
1. The OSM Api should reject creating a note without an initial comment
2. Either all current notes with no comments are deleted from DB or displaying a note with no comment should not yield an error 500

**Comment by tomhughes** [comment]
I don't think that's true is it. I thought @Firefishy established that the cause was that the creator of the note had been deleted because they were a spammer?

**Comment by tomhughes** [comment]
So the immediate cause might be that there are no (visible) comments, but the conclusion that the API allows creation of a note without a comment is not - the only bug here is the exception when a note has no visible comments.

**Comment by westnordost** [comment]
That might as well be, I can't see whether it is invisible or never existed, I only reported what I saw. Anyway, if the (only) comment was deleted, then error 500 would still not be appropriate.

**Comment by tomhughes** [comment]
Sorry I assumed it was you that had been discussing it with @Firefishy on IRC last night but I now see that it wasn't and in fact that was a different note.

**Comment by tomhughes** [comment]
The cause is essentially the same - the user that opened this note deleted their account causing their note comments to be hidden.

This wasn't a spam note though, which raises the question of whether we should in fact hide comments from deleted users or not, or whether we should just treat them as anonymous,

**Comment by westnordost** [comment]
I guess a good approach would be to treat it the same as map data and GPS traces a deleted user contributed.

**Comment by Dimitar5555** [comment]
This issue is still present.
https://api.openstreetmap.org/api/0.6/notes/2523764
https://www.openstreetmap.org/note/2523764

**Comment by tomhughes** [comment]
That would be why this issue is still open yes. Did have something useful to add?

**Comment by Dimitar5555** [comment]
The note that westnodorst linked seems to no longer results in a HTTP 500 error and currently it has content (i.e. it's not empty).

**Comment by Discostu36** [comment]
Sorry for not finding this issue before opening mine. I think you should hide/delete these notes. There is no use for empty notes leading to error pages, it only makes maintenance harder.

**Comment by tomhughes** [comment]
We already have a plan (#3831) to fix this which we have started work on.

**Comment by kmpoppe** [comment]
@tomhughes by skimming the planet-notes.osn I found [1555586](https://api.openstreetmap.org/api/0.6/notes/1555586.json) and [1555588](https://api.openstreetmap.org/api/0.6/notes/1555588.json) which seem to have no public comments. They're from 2018, are those a symptom of the original user having deleted their account?

**Comment by pa5cal** [comment]
> @tomhughes by skimming the planet-notes.osn I found [1555586](https://api.openstreetmap.org/api/0.6/notes/1555586.json) and [1555588](https://api.openstreetmap.org/api/0.6/notes/1555588.json) which seem to have no public comments. They're from 2018, are those a symptom of the original user having deleted their account?

Yes, the note was hidden, perhaps because of its text content or the original user was deleted.
Then someone else closed it.
